src/services/fetcher.tsx

```diff
@@ -1 +1,42 @@
-export {}
+import axios, { AxiosRequestConfig, AxiosResponse } from "axios";
+
+import { OperationResult } from "../interfaces/IOperationResult";
+
+export abstract class Fetcher {
+    constructor(){
+        this.initializeRequestInterceptors();
+    }
+
+    private initializeRequestInterceptors = () => {
+        axios.interceptors.request.use(
+            this.handleRequest,
+            this.handleError
+        )
+    }
+
+    private handleRequest = (config:AxiosRequestConfig) => {
+        if(this.token){
+            config.headers = {
+                'Authorization': `Bearer ${this.token}`,
+                ...config.headers ?? {}
+            }
+        }
+        return config;
+    }
+    private handleError = (error: any) => Promise.reject(error);
+    private readonly token = localStorage.getItem('access_token');
+
+    async get<T>(url: string, isProtected=true):Promise<AxiosResponse<OperationResult<T>>> {
+        let config:AxiosRequestConfig = {};
+        if(isProtected) {
+            //config.headers.Authorization = 'Bearer ${this.token}';
+            config = {headers: {'Authorization': `Bearer ${this.token}`}}
```

Comment:
Je pense que tu peux le retirer également, c est géré dans le interceptor normalement

Reply:
Alors du coup petite question, je vire config des paramètres de get et post, du coup ?

Reply:
Tu peux laisser pour le moment au cas où on aura d'autres headers ou d'autres config spécifiques à passer mais tu peux retirer la ligne             config = {headers: {'Authorization': `Bearer ${this.token}`}} du get

Reply:
Ok, je te laisse jeter un coup d'oeill sur le prochain commit.